src/widget/scenario_minimap.c

```diff
@@ -0,0 +1,144 @@
+            {0x6b8431, 0x737b31}, {0x6b7b31, 0x737b29}, {0x636b18, 0x526b21}, {0x737b31, 0x737b29}
+        },
+        .road = {0x736b63, 0x4a3121}
+    },
+    // northern
+    {
+        .water = {{0x394a7b, 0x31427b}, {0x394a7b, 0x314273}, {0x313973, 0x314273}, {0x31427b, 0x394a7b}},
+        .tree = {{0x527b31, 0x082108}, {0x083908, 0x5a7329}, {0x082908, 0x316b21}, {0x527b29, 0x084a21}},
+        .rock = {{0x8c8484, 0x5a5252}, {0x9c9c94, 0xa5a5a5}, {0xa5a5a5, 0x848484}, {0x5a5252, 0x9c9c94}},
+        .meadow = {{0x427318, 0x8c9442}, {0xb5ad4a, 0x738c39}, {0x8c8c39, 0x6b7b29}, {0x527331, 0x5a8442}},
+        .grass = {
+            {0x4a8431, 0x4a7329}, {0x527b29, 0x4a7329}, {0x526b29, 0x5a8439}, {0x397321, 0x4a6b21},
+            {0x527b31, 0x5a7331}, {0x4a7329, 0x5a7329}, {0x4a6b18, 0x316b21}, {0x527b29, 0x527329}
+        },
+        .road = {0x736b63, 0x4a3121}
+    },
+    // desert
+    {
+        .water = {{0x4a84c6, 0x4a7bc6}, {0x4a84c6, 0x4a7bc6}, {0x4a84c6, 0x5284c6}, {0x4a7bbd, 0x4a7bc6}},
+        .tree = {{0xa59c7b, 0x6b7b18}, {0x214210, 0xada573}, {0x526b21, 0xcec6a5}, {0xa59c7b, 0x316321}},
+        .rock = {{0xa59494, 0x736352}, {0xa59c94, 0xb5ada5}, {0xb5ada5, 0x8c847b}, {0x736352, 0xbdada5}},
+        .meadow = {{0x739c31, 0x9cbd52}, {0x7bb529, 0x63ad21}, {0x9cbd52, 0x8c944a}, {0x7ba539, 0x739c31}},
+        .grass = {
+            {0xbdbd9c, 0xb5b594}, {0xc6bda5, 0xbdbda5}, {0xbdbd9c, 0xc6c6ad}, {0xd6cead, 0xc6bd9c},
+            {0xa59c7b, 0xbdb594}, {0xcecead, 0xb5ad94}, {0xc6c6a5, 0xdedebd}, {0xcecead, 0xd6d6b5}
+        },
+        .road = {0x6b5a52, 0x4a4239}
+    }
+};
+
+static struct {
+    int absolute_x;
+    int absolute_y;
+    int width_tiles;
+    int height_tiles;
+    int x_offset;
+    int y_offset;
+} data;
+
+static void foreach_map_tile(map_callback *callback)
+{
+    city_view_foreach_minimap_tile(
+        data.x_offset, data.y_offset, data.absolute_x, data.absolute_y,
+        data.width_tiles, data.height_tiles, callback);
+}
+
+static void set_bounds(int x_offset, int y_offset, int width, int height)
+{
+    data.width_tiles = width / 2;
+    data.height_tiles = height;
+    data.x_offset = x_offset;
+    data.y_offset = y_offset;
+    data.absolute_x = (VIEW_X_MAX - data.width_tiles) / 2;
+    data.absolute_y = (VIEW_Y_MAX - data.height_tiles) / 2;
+
+    // ensure even height
+    data.absolute_y &= ~1;
+}
+
+static void draw_minimap_tile(int x_view, int y_view, int grid_offset)
+{
+    if (grid_offset < 0) {
+        return;
+    }
+
+    int terrain = map_terrain_get(grid_offset);
+
+    if (terrain & TERRAIN_BUILDING) {
+        // Native huts/fields
+        if (map_property_is_draw_tile(grid_offset)) {
+            int image_id = image_group(GROUP_MINIMAP_BUILDING);
+            switch (map_property_multi_tile_size(grid_offset)) {
+                case 1: image_draw(image_id, x_view, y_view); break;
```

Comment:
Instead of `case N:`, you could use:

```
int offset = map_property_multi_tile_size(grid_offset) - 1;
image_draw(image_id + offset, x_view, y_view - offset);
```

Reply:
That will lead to problems when buildings ever get bigger than 5x5 (Augustus...). In fact, for the scenario minimap we'll only ever need 1x1 and 2x2 so I'll get rid of the other cases.